youtube_dl/extractor/piksel.py

```diff
@@ -73,10 +74,9 @@ def _extract_url(webpage):
         if mobj:
             return mobj.group('url')
 
-    def _call_api(self, app_token, resource, display_id, query, fatal=True):
-        response = (self._download_json(
-            'http://player.piksel.com/ws/ws_%s/api/%s/mode/json/apiv/5' % (resource, app_token),
-            display_id, query=query, fatal=fatal) or {}).get('response')
+    def _call_api(self, app_token, resource, display_id, query, host="https://player.piksel.com", fatal=True):
+        url = urljoin(base=host, path='/ws/ws_%s/api/%s/mode/json/apiv/5' % (resource, app_token))
+        response = (self._download_json(url, display_id, query=query, fatal=fatal) or {}).get('response')
```

Comment:
We could ensure that a `dict` is always returned, even if the returned JSON isn't one (add `traverse_obj` to utils imports). Also, no need to use kwargs for `urljoin()`:
```suggestion
        url = urljoin(host, '/ws/ws_%s/api/%s/mode/json/apiv/5' % (resource, app_token))
        response = self._download_json(url, display_id, query=query, fatal=fatal)
        response = traverse_obj(response, 'response', expected_type=dict) or {}
```